analyzers/src/SonarAnalyzer.CFG/CfgSerializer/CfgSerializer.RoslynLvaWalker.cs

```diff
@@ -36,9 +36,21 @@ public RoslynLvaWalker(RoslynLiveVariableAnalysis lva, DotWriter writer, RoslynC
 
         protected override void WriteEdges(BasicBlock block)
         {
-            foreach (var predecessor in lva.BlockPredecessors[block.Ordinal])
+            var lvaPredecessors = lva.BlockPredecessors[block.Ordinal];
```

Comment:
There's a lot of duplication now :/ 
Maybe it's possible to change what you need here and then call the base.WriteEdges?
Maybe you can also give the predecessors are input?